drivers/virtio/net/virtio_net.c

```diff
@@ -1291,10 +1322,13 @@ static void virtio_net_info_get(struct uk_netdev *dev,
 				struct uk_netdev_info *dev_info)
 {
 	struct virtio_net_device *vndev;
+	__u64 host_features;
 
 	UK_ASSERT(dev && dev_info);
 	vndev = to_virtionetdev(dev);
 
+	host_features = virtio_feature_get(vndev->vdev);
```

Comment:
Why not use the cached features from `vndev->vdev->features`?

Reply:
That field contains the already negotiated features, not the features that are available. For `info_get` the client wants to know what features it could enable (some features such as LRO might be not supported). Afterwards, it will call the configure function and in there the "client optional" features are then set in `vndev->vdev->features`.